Adding sessionCloseCallbacks

mourad opened the pull request.

Commit:
Adding sessionCloseCallbacks

jcelliott commented:
Thanks. Makes sense to add this. I'm also planning on implementing the WAMP (advanced profile) [Session Meta API](https://github.com/tavendo/WAMP/blob/master/spec/advanced/session-meta-api.md). Would that fill the same use-case you wanted this for, or would you still want to have this "embedded" access?

mourad commented:
I like the idea of the Session Meta API.  I have not yet implemented the usage of the session close callback that I added, but I added it here for completeness ( to mirror the session open callbacks ) as their use will come up in the future.  The Session Meta API on_join/on_leave should be sufficient, so when you get to implement it, feel free to remove the callbacks.
